core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -93,10 +96,22 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
         return hashCode();
       } else if ("toString".equals(method.getName())) {
         return toString();
+      } else if (Util.isDefault(method)) {
+        return invokeDefault(proxy,method, args);
       }
       return dispatch.get(method).invoke(args);
     }
 
+    private Object invokeDefault(Object proxy, Method method, Object[] args) throws Throwable {
+      Class<?> declaringClass = method.getDeclaringClass();
+      Field field = Lookup.class.getDeclaredField("IMPL_LOOKUP");
+      field.setAccessible(true);
+      Lookup lookup = (Lookup) field.get(null);
+      return lookup.unreflectSpecial(method, declaringClass)
```

Comment:
benchmarking might find this method is expensive. if so, we might try to cache the MethodHandle lookups in the ctor, similar to `dispatch`. Ack this might be tricky to have work in JRE 6.

Ex you might have to use a field like `Map<Method, Object> defaultMethods`, and then do a containsKey on it and cast out.. like..

```java
} else if (defaultMethods.containsKey(method)) {
  return invokeDefault(method);
}
--snip--

@IgnoreJRERequirement // Should only be invoked on Java 7+
Object invokeDefault(Method defaultMethod) throws Exception {
  MethodHandle handle = (MethodHandle) defaultMethods.get(defaultMethod);
  return handle.invokeWithArguments(args);
}
...


```